### ISSUE TITLE
Notification that a case has been seen by a given user

### ISSUE THREAD

**Issue by franck-boullier** [unee-t/frontend#191]
# The need:
We need a way to display the information that the message has been read by a given user.

This should be relatively easy to do using the BZ 'tag' mechanism:
1- it is possible for any user to add a tag to any message or comment
2- tag can be anything 

# Idea for an implementation:
1- The MEFE creates a BZ tag (user_id + timestamp) each time a user accesses the conversation via the MEFE. This tag is attached to the all the new messages since the last user visit (all the messages with out a tag which include that user_id)
2- The MEFE interface displays this tag in a way that the users can understand:
  2.1- A single grey tick to signal that the message is being sent
  2.3- A single colored tick to signal that the message has been sent to the BZ db.
  2.2- A second grey tick to signal that the message has been seen by some users
  2.3- A second colored tick to signal that the message has been seen by all invited users
  2.4- When the user long press on the message, possibility to access readable information like User [MEFE user name] has seen this message on [timestamp]

@nbiton, do you see any issue with this approach?
@kiatlim any recommendation on the design for that? Could you think of how we would display that in the interface?

**Comment by kaihendry** [comment]
We effectively need to keep a log when a user accessed said content.

I think the most important first step is to agree how each piece of content is identified.

https://dev.dashboard.unee-t.com/show_bug.cgi?id=61918#c4

So that is, case number + sequence.

Or do we go by the DB's comment ID?

<img src=https://s.natalian.org/2018-05-23/1527045476_2558x1406.png alt="comment id">

http://localhost:8082/?server=auroradb.dev.unee-t.com&username=root&db=bugzilla&select=longdescs&columns%5B0%5D%5Bfun%5D=&columns%5B0%5D%5Bcol%5D=&where%5B0%5D%5Bcol%5D=bug_id&where%5B0%5D%5Bop%5D=%3D&where%5B0%5D%5Bval%5D=61918&where%5B01%5D%5Bcol%5D=&where%5B01%5D%5Bop%5D=%3D&where%5B01%5D%5Bval%5D=&order%5B0%5D=&limit=50&text_length=100

**Comment by kaihendry** [comment]
Oh, I just noticed [bug_user_last_visit](http://localhost:8082/?server=auroradb.dev.unee-t.com&username=root&db=bugzilla&select=bug_user_last_visit) which tells us when the user last visited the bug in Bugzilla.

Though I am not 100% sure if it updates if only accessed via the frontend.

<img src=https://s.natalian.org/2018-05-23/1527047854_2558x1406.png>

**Comment by franck-boullier** [comment]
> We effectively need to keep a log when a user accessed said content.

@kaihendry  this has already been implemented: all these information is in the table 'ut_notification_messages_cases'
It is automatically updated with the information we need in real time

**Comment by kaihendry** [comment]
Oh, well I didn't know this...

<img src="https://s.natalian.org/2018-05-23/1527054299_2558x1406.png">

I don't see how that links to a particular comment/longdesc. All it shows is activity, like table=bugs_activity ? Sorry I thought "message" meant an individual update on a case. Not the whole case?

**Comment by franck-boullier** [comment]
> I thought "message" meant an individual update on a case

Nope: Message means, a new message has been added to that case by that specific user.

> I don't see how that links to a particular comment/longdesc.

It does not and it does not need to: this is notification, we do NOT to get the details here, just the information that something happened and generic information about the type of thing that happened.

**Comment by kaihendry** [comment]
Still a little confused as to what is the source of truth of how we determine that a user saw (not interacted, for that we have [engagement](https://github.com/unee-t/engagement)) a message: https://github.com/unee-t/frontend/issues/331#issuecomment-463971683

Perhaps given a userId and notificationId, what is the SQL (or Mongo) query to determine it was read?
